date/index.js

```diff
@@ -1,4 +1,22 @@
 import moment from 'moment';
+import 'moment-timezone';
+import 'moment-timezone/moment-timezone-utils';
+
+export const settings = window._wpDateSettings;
+
+// Create WP timezone based off dateSettings.
+const offsets = ( settings.timezone.string ) ? moment.tz.zone( settings.timezone.string ).offsets : [ -settings.timezone.offset * 60 ];
+const momentTimezone = {
+	name: 'WP',
+	abbrs: [ 'WP' ],
+	untils: [ null ],
+	offsets: [ offsets ],
+};
+moment.tz.add( moment.tz.pack( momentTimezone ) );
+
+// Create a new wpmoment object attaching the timezone and cloning
+// so it does not change the global moment timezone.
+export const wpmoment = moment.tz( 'WP' ).clone();
```

Comment:
Instead of exporting this, maybe we could proxify the calls to moment instead.

```js
import momentLibrary from 'moment';

export const moment = (...args) => {
  return momentLibrary.tz(...args, 'WP');
};
```
not certain it works with all the different ways to instanciate moment though

Reply:
Also, I forgot I did attempt something similar but ran into issues because the moment timezone library attaches itself to the moment namespace, so renaming the library on import doesn't work. 

It is why I ended up with wpmoment name, though I think I prefer wpdate.